### PR TITLE
(0.90.11) Tests for a distributed Immersed boundary grid + fix some `on_architecture` bugs

### PR DESCRIPTION
**Pull request by simone-silvestri**
This PR introduces a test to verify the correct execution of a distributed immersed latitude longitude grid

### CODE REVIEW

File: test/test_distributed_hydrostatic_model.jl [hunk 1]
```diff
@@ -78,49 +78,55 @@ Ny = 32
...
+            ηs = partition_global_array(cpu_arch, ηs, size(η))
+
+            atol = eps(eltype(grid))
+            rtol = sqrt(eps(eltype(grid)))
+
+            @test all(isapprox(u, us; atol, rtol))
+            @test all(isapprox(v, vs; atol, rtol))
+            @test all(isapprox(w, ws; atol, rtol))
+            @test all(isapprox(c, cs; atol, rtol))
+            @test all(isapprox(η, ηs; atol, rtol))
```
Comment (glwagner) [inline test/test_distributed_hydrostatic_model.jl:129]: Is is possible to write this test to work with `Field` and `on_architecture` and avoid using `Array`?

Comment (simone-silvestri) [reply test/test_distributed_hydrostatic_model.jl:129]: first, we need an `on_architecture` method for fields. I was thinking already about it but I think it should be done in another PR.

Comment (navidcy) [reply test/test_distributed_hydrostatic_model.jl:131]: Let's wait for #3490 to be merged then?

**Comment by navidcy** [comment]
@simone-silvestri, have a look at this error when you find time:
https://buildkite.com/clima/oceananigans-distributed/builds/1464#018e1b66-735b-43bc-9c77-4bb7b4646c2d/192-4782

**Review by navidcy — APPROVED** [review]

**Comment by simone-silvestri** [comment]
Thanks @navidcy. I ll merge when tests pass

**Pull request merged**